### ISSUE TITLE
Include delivery address id in live integration order

### ISSUE THREAD

**Issue by dwlhs** [dynamicweb/DynamicWeb#130]
Hi :)

When B2B customers submit an order with an alternative delivery address selected, the internal AccessUserAddressId is stored on the order. In the integration, however, only the address information (street, city, e.t.c.) is transferred and not the code of the address. The address record in Business Central can carry information on how goods are shipped to this address, which is lost on the sales order when the address is just filled in:

![image](https://github.com/user-attachments/assets/bb6593c7-8eec-4e5f-94ee-75f9853d5a13)

To trigger this information on the orders, please add the ERP native address code to the integration. Thanks :)

Br.
Lars

**Comment by DynamicRas** [comment]
Hi, 

What is it exactly you would like to have stored and where? You mention live integration in the topic, but aren't you storing the address on the user? And in the topic you mention delivery address id and in the text you want delivery address code. 

BR Rasmus Sanggaard

**Comment by dwlhs** [comment]
Hi Rasmus,

In short terms, I want the BC "code" of the selected address on the order in DW stamped on the sales order in BC, when DW sends the order.

When we import customers and addresses, the addresses are imported in relation to the user. In this process, we set the address "code" from BC (i.e., the record identifier of the address in BC) as AccessUserAddressUid, so that on subsequent imports of addresses we can identify if an address already existes in DW or it's a new one. Users can have multiple addresses, and they can select delivery for any of these addresses.

BUT... When we send orders to BC, we don't tell BC which address record we want the goods delivered to. We only send the address information. By specifying the address record, BC will be aware of the terms of delivery to that address..

SO when sending the order to BC, I would like the value of the AccessUserAddressUid to be included in the XML and saved on the sales header in BC.

Please note that the imported AccessUserAddressUid is a concatenation of the contact code and address code, so the address code needs to be separated from this to be recognized in BC.

Br.
Lars

**Comment by nicped** [comment]
Hi Lars
Can Order.CustomerAddressId and Order.DeliveryAddressId be used in this case? (They are relatively new properties made for the same reason)

BR Nicolai

**Comment by dwlhs** [comment]
Hi Nicolai,

They can be used to refer to the AccessUserAddressId in AccessUserAddress, but for recording the address on the sales header in BC, we need part of the value in AccessUserAddressUid. A picture says more than a thousand words :)

![image](https://github.com/user-attachments/assets/27a32694-150b-454a-8123-c3bf05ee525e)

It's the "AddressCodeToBC" on the picture that should be recorded on the sales header in BC.

Br.
Lars

**Comment by nicped** [comment]
Thanks. I might be stupid - this is an extremely confusing thread to understand.... :-).

What field from the address in BC do you want on the order when it comes back to BC from DW? Is it the "Code" field?

**Comment by dwlhs** [comment]
Well, that's one opinion :)

I don't want anything new coming back to DW. This is related to the order that DW sends to BC.

Today, when we send an order to BC, the address is recorded on the sales header as "Custom address":

![image](https://github.com/user-attachments/assets/2d6d44fe-8ebd-4c92-9b7c-2169786b9d09)

This is just information on where to send the goods.

I want us to include the address "code" (see below), because then the order will be recorded with "Alternative Shipping Address", and the address is then selected from the list of addresses recorded on the customer in BC:

![image](https://github.com/user-attachments/assets/3b38e3a2-b1c1-469b-bf14-499e5fccbb70)

The reason for this is that the address record in BC contains information about, where the goods should be shipped **from**, how freight costs are charged, which carrier e.t.c. according the standard agreement with the customer:

![image](https://github.com/user-attachments/assets/736c3a5a-dce4-4c98-a985-9552877b3498)

We get this code when importing customers and contacts and store it in AccessUserAddressUid. But because we don't have a concept, where contact persons can share addresses, we prefix the address code with contactid and repeat import the same address for all contacts in a company. So, in order to send the right code to BC, we need to isolate it from the AccessUserAddressUid as in the SQL example in the previous post.

Hope it makes sense :)

Br.
Lars

**Comment by nicped** [comment]
Yes and no.

I now understand what you want. But we also need another solution than what you suggest.

Could it be something like this?
- Add code to address table so it can be mapped and retrieved directly. 
- Consider other address fields from BC/ERP side
- Make sure contacts can share addresses on their "account" using e.g customer number when we lookup the table (Team DWNA have same request)
- Move code from our address table to a new field on the order for address code (and maybe other fields)
- Receive and map addresscode from BC to DW address table
- Send order.deliveryaddresscode from DW to BC

**Comment by dwlhs** [comment]
Yep, that pretty much wraps it up, and shared addresses makes it even better :)

**Comment by DynamicRas** [comment]
LocationCode, ShipmentMethodCode, ShippingAgentCode and ShippingAgentServiceCode to the AccessUserAddress table
